File: Examen_Corto/Python/Ejercicio4.py
import psycopg2

numero = int(input("Ingrese un número: "))

suma = sum(range(1, numero + 1))
numeros = "+".join(map(str, range(1, numero + 1)))

salida = f"Numero {numero} : {numeros} = {suma}"
print(salida)

# Escribir en el archivo
with open("ejercicio7.txt", "w") as file:
    file.write(salida)

# Conexión a la base de datos PostgreSQL
try:
    conn = psycopg2.connect(
        dbname='ExamenCorto1',
        user='postgres',
        password='gabrielgrdb',
        host='localhost',
        port='5432'
    )

    cursor = conn.cursor()

    query = "insert into salida values (%s);"
    cursor.execute(query, (salida,))

    conn.commit()

except Exception as e:
    print("Error de conexión:", e)

finally:
    if conn is not None:
        conn.close()

File: Examen_Corto/Octave/ejercicio4.m


numero = input("Ingrese un número: ");

suma = sum(1:numero);
numeros = strjoin(arrayfun(@num2str, 1:numero, "UniformOutput", false), "+");

salida = ["Numero " num2str(numero) " : " numeros " = " num2str(suma)];
disp(salida);


fid = fopen("ejercicio7.txt", "w");
fprintf(fid, "%s", salida);
fclose(fid);

% Conexión a la base de datos PostgreSQL
pkg load database %cargar el paquete

conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
'port','5432','user','postgres','password','gabrielgrdb'))


query = sprintf("insert into salida values ('%s');", salida);
N = pq_exec_params(conn, query);
